http/rest-client-reactive/src/main/java/io/quarkus/ts/http/restclient/reactive/UrlOverrideableClient.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.ts.http.restclient.reactive;
+
+import java.net.URI;
+import java.net.URL;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+
+import io.quarkus.rest.client.reactive.Url;
+
+/**
+ * RestClient which provides capability to override base URL.
+ * URL might be overridden to access different service.
+ * Base URL can be overridden by String, URL or URI parameter type.
+ */
+@RegisterRestClient
+@Path("urlOverride")
+public interface UrlOverrideableClient {
+
+    @GET
+    @Path("basic")
+    String getByString(@Url String uri);
+
+    @GET
+    @Path("basic")
+    String getByUrl(@Url URL uri);
+
+    @GET
+    @Path("basic")
+    String getByUri(@Url URI uri);
```

Comment:
You override `@Url` in all the three client methods, how do you know that non-overriden method still works? Maybe consider testing it.

Reply:
I'm testing it. If you pass "null" as an argument, it leaves it non-overriden. (I took this approach from unit tests for the feature itself).

Reply:
That's not a same as if you have a REST client method without `@Url` though, it depends on the implementation. Nevertheless, I leave it to you to decide if you want to test it...

Reply:
I've added another method and test to cover non-override case.